modules/core/src/hal_replacement.hpp

```diff
@@ -98,6 +98,19 @@ inline int hal_ni_sub64f(const double *src1_data, size_t src1_step, const double
 
 inline int hal_ni_sub8u32f(const uchar *src1_data, size_t src1_step, const uchar *src2_data, size_t src2_step, float *dst_data, size_t dst_step, int width, int height) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
 inline int hal_ni_sub8s32f(const schar *src1_data, size_t src1_step, const schar *src2_data, size_t src2_step, float *dst_data, size_t dst_step, int width, int height) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
+
+/**
+Add scalar: _dst[i] = src[i] + scalar
+
+@param src_data source image data
+@param src_step source image step
+@param dst_data destination image data
+@param dst_step destination image step
+@param width width of the images
+@param height height of the images
+@param scalar_data pointer to scalar value
+*/
+inline int hal_ni_addScalar32f32f(const float *src_data, size_t src_step, float *dst_data, size_t dst_step, int width, int height, const float* scalar_data) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
if we want to support C1, C3 etc. scalars, we need to add `int nchannels` (check that the name is consistent with other HAL entries)

Reply:
Probably we will add support for more channels in later PRs
For now we leave it just for 1 channel

Reply:
ok